sirepo/package_data/static/js/raydata.js

```diff
@@ -377,11 +380,14 @@ SIREPO.app.directive('scansTable', function() {
             $scope.images = null;
             $scope.noScansReturned = false;
             $scope.orderByColumn = 'start';
+            $scope.pdfSelectAllScans = false;
+            $scope.pdfSelectedScans = [];
             $scope.reverseSortScans = false;
             $scope.runLogModalId = 'sr-view-log-iframe-' + $scope.analysisStatus;
             $scope.runLogScanId = null;
             $scope.scans = [];
             $scope.selectedScan = null;
+            $scope.showPdfColumn = $scope.analysisStatus === 'executed';
```

Comment:
This addition made me look at this entire directive and wonder if there is a way to split it up. The controller is 267 lines long which is big enough to be fairly unwieldy. We don't have to do it now but as you are working in the code it would be good to keep in the back of your mind how to split things up.

One idea I have is to split some of the business logic and presentation logic up. For example, we could have a wrapping directive that knows how to get the scans (sendScanRequest). Then we could pass the scans into a directive that knows how to present them (show the table, sort the columns, etc). 

I don't have a good idea how to separate out things like the pdf. Only some tables want it so it would be nice to separate out the logic/presentation for it into something that knows just how to deal with pdfs. Then it could be injected/attached to the scans table in that instance. Angular doesn't provide much in the way for doing things like this...